### PR TITLE
React client ui firebase integration

### PR DESCRIPTION
**Pull request by prabhuinbarajan**
firebase auth integration and environment refactoring

### CODE REVIEW

**Comment by therealnikhil** [comment]
Quite a few questions/comments:
- Why create a new login component when component already exists?
- We have collectively decided to use React Hooks and not class components
- `npm run dev` no longer serves localhost:8080, so where am I supposed to see the login page to test this?

**Comment by therealnikhil** [comment]
Use `admin@soylentgreen.com` to test

File: client/src/components/Firebase/firebase.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+
+
+onAuthUserListener = (next, fallback) =>
+    this.auth.onAuthStateChanged(authUser => {
+      if (authUser) {
+             this.getUserDocument(authUser.email).then(userDoc => {
+                if (!userDoc.roles) {
+                  userDoc.roles = {
+                    ADMIN: userDoc.isAdmin,
```
[the remark this answers is not in the document]
Comment (prabhuinbarajan) [reply client/src/components/Firebase/firebase.js:85]: @ibeckermayer  - i just tested this to make sure this works. But this is not internally used in login today. we are using the JWT token for login

const User = types
  .model({
    isAdmin: types.boolean,
  })
  .actions((self) => {
    const signIn = flow(function* (firebase, email, password) {
      try {
        yield firebase.doSignInWithEmailAndPassword(email, password)
        console.log('Successfully logged in')

        const idTokenResult = yield firebase.auth.currentUser.getIdTokenResult(true)

        self.isAdmin = idTokenResult.claims.isAdmin
        console.log(`isAdmin set to ${self.isAdmin}`)
      } catch (e) {
        console.log(e)
      }
    })

    return {
      signIn,
    }
  })

File: client/src/components/Firebase/firebase.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import app from 'firebase/app';
+import "firebase/auth";
+import { functions } from "firebase";
+import * as firebaseConfigLocal from '../../config/firebase.config.local';
+import * as firebaseConfigDev from '../../config/firebase.config.dev';
+import * as firebaseConfigTest from '../../config/firebase.config.test';
+import * as firebaseConfigProd from '../../config/firebase.config.prod';
+
+//require('firebase/auth')
```
[the remark this answers is not in the document]
Comment (prabhuinbarajan) [reply client/src/components/Firebase/firebase.js:9]: fixed

File: client/src/components/Firebase/firebase.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibeckermayer) [reply client/src/components/Firebase/firebase.js:83]: Right, but the point of that `isAdmin` previously was to protect a route. Now we're keeping track of it in two different places, in two different ways -- is that necessary? Or can we make it simpler by just choosing one of them?

On a related note, when exploring this issue I noticed we have these two functions in `Login.v1.js` which appear to be doing the same thing, although I'm not sure that the latter is ever being called. Can we clean that up?

```javascript
clickSubmit = async (event) => {
    event.preventDefault()
    //this.setValues({ ...values, error: false })
    console.log(`isAdmin = ${store.user.isAdmin}`)
    const { email, password } = this.state;


    try {
      await store.user.signIn(this.props.firebase, email, password);
      if (store.user.isAdmin === true) {
        this.props.history.push(ROUTES.MANAGE_MEMBERS);
      }
      //setValues({ ...values, redirect: store.user.isAdmin })
    } catch (err) {
      console.log(err)
      setValues({ ...values, redirect: store.user.isAdmin })
    }
  }


  onSubmit = event => {
    event.preventDefault();
    const { email, password } = this.state;
    this.props.firebase
      .doSignInWithEmailAndPassword(email, password)
      .then((user) => {
        this.setState({ ...INITIAL_STATE });
        this.props.firebase.generateUserDocument(user.user).then( userDoc => { 
          if (userDoc.isAdmin === true) {
            this.props.history.push(ROUTES.MANAGE_MEMBERS);
          }
        })
      })
      .catch(error => {
        this.setState({ error });
      });

    event.preventDefault();
  };
```

**Pull request merged**